streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -29,6 +32,37 @@ trait ZSinkPlatformSpecificConstructors { self: ZSink.type =>
         case e: IOException => e
       }
     }
+
+  /**
+   * Uses the provided `Path` to create a [[ZSink]] that consumes byte chunks
+   * and writes them to the `File`. The sink will yield count of bytes written.
+   */
+  final def fromFile(
+    path: => Path,
+    position: Long = 0L,
+    options: Set[OpenOption] = Set(WRITE, TRUNCATE_EXISTING, CREATE)
+  ): ZSink[Blocking, Throwable, Byte, Long] =
+    ZSink {
+      for {
+        state <- Ref.make(0L).toManaged_
+        channel <- ZManaged.make(
+                    blocking.effectBlockingInterrupt(FileChannel.open(path, options.asJava).position(position)).orDie
+                  )(chan => blocking.effectBlocking(chan.close()).orDie)
+        push = (is: Option[Chunk[Byte]]) =>
+          is match {
+            case None => state.get.flatMap(Push.emit)
+            case Some(byteChunk) =>
+              for {
+                bytesWritten <- state.get
+                justWritten <- blocking.effectBlockingInterrupt {
+                                Files.write(path, byteChunk.toArray)
+                                channel.write(ByteBuffer.wrap(byteChunk.toArray))
+                              }.mapError(Left(_))
+                more <- state.set(bytesWritten + justWritten) *> Push.more
```

Comment:
You could save the first `state.get` by doing `state.update(_ + justWritten)`.

Reply:
Got it! 👌